Makefile

```diff
@@ -0,0 +1,61 @@
+GOLANGCI_VERSION ?= 'latest'
+
+##@ General
+
+# The help target prints out all targets with their descriptions organized
+# beneath their categories. The categories are represented by '##@' and the
+# target descriptions by '##'. The awk commands is responsible for reading the
+# entire set of makefiles included in this invocation, looking for lines of the
+# file as xyz: ## something, and then pretty-format the target and help. Then,
+# if there's a line with ##@ something, that gets pretty-printed as a category.
+# More info on the usage of ANSI control characters for terminal formatting:
+# https://en.wikipedia.org/wiki/ANSI_escape_code#SGR_parameters
+# More info on the awk command:
+# http://linuxcommand.org/lc3_adv_awk.php
+
+.PHONY: help
+help: ## Display this help.
+	@awk 'BEGIN {FS = ":.*##"; printf "\nUsage:\n  make \033[36m<target>\033[0m\n"} /^[a-zA-Z_0-9-]+:.*?##/ { printf "  \033[36m%-15s\033[0m %s\n", $$1, $$2 } /^##@/ { printf "\n\033[1m%s\033[0m\n", substr($$0, 5) } ' $(MAKEFILE_LIST)
+
+
+
+##@ Development
+
+.PHONY: test
+TEST_ARGS ?= -v
+TEST_TARGETS ?= ./...
+test: lint ## Test the Go modules within this package.
+	@ echo ▶️ go test $(TEST_ARGS) $(TEST_TARGETS)
+	go test $(TEST_ARGS) $(TEST_TARGETS)
+	@ echo ✅ success!
+
+
+.PHONY: lint
+LINT_TARGETS ?= ./...
+lint: golangci-lint ## Lint Go code with golangci-lint
+	@ echo "▶️ golangci-lint run"
+	$(GOLANGCILINT) run $(LINT_TARGETS)
+	@ echo "✅ golangci-lint run"
+
+
+##@ Build Dependencies
+
+## Location to install dependencies to
+LOCALBIN ?= $(shell pwd)/bin
+$(LOCALBIN):
+	mkdir -p $(LOCALBIN)
+
+
+##@ Download utilities
+
+OS := $(shell go env GOOS)
+ARCH := $(shell go env GOARCH)
+
+.PHONY: golangci-lint
+GOLANGCILINT := $(LOCALBIN)/golangci-lint
+GOLANGCI_URL := https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh
+golangci-lint: $(GOLANGCILINT) ## Download golangci-lint
+$(GOLANGCILINT): $(LOCALBIN)
+	@ echo "▶️ Downloading golangci-lint"
+	curl -sSfL $(GOLANGCI_URL) | sh -s -- -b $(LOCALBIN) $(GOLANGCI_VERSION)
```

Comment:
There are different opinions on the `curl | sh` pattern, I mostly think of it as an antipattern. Can we remove downloading here and just error out if golangci-lint is not present?

Reply:
Note: we are also using the golangci-lint [action](https://github.com/golangci/golangci-lint-action),